app/src/services/sync.js

```diff
@@ -14,41 +14,87 @@ const versionService = require('./version');
+        objId = s3ObjectComsId;
+      } else { // Update S3 Object if there is still remaining space in the TagSet
+        if (TagSet.length < 9) { // putObjectTagging replaces S3 tags so new TagSet must contain existing values
+          await storageService.putObjectTagging({
+            filePath: path,
+            bucketId: bucketId,
+            tags: TagSet.concat([{ Key: 'coms-id', Value: objId }])
+          });
+        }
+      }
+    } else if (typeof s3Object === 'boolean') { // If soft-deleted S3 Object
+      const { Versions } = await storageService.listAllObjectVersions({ filePath: path, bucketId: bucketId });
+
+      for (const versionId of Versions.map(version => version.VersionId)) {
+        const result = await storageService.getObjectTagging({
+          filePath: path,
+          s3VersionId: versionId,
+          bucketId: bucketId
+        });
+        const oldObjId = result?.TagSet.find(obj => obj.Key === 'coms-id')?.Value;
+
+        if (oldObjId && uuidValidate(oldObjId)) {
+          objId = oldObjId;
+          break; // Stop iterating if a valid uuid was found
+        }
+      }
+    }
+
+    return Promise.resolve(objId);
+  },
+
   /**
    * @function syncJob
    * Orchestrates the synchronization of all aspects of a specified object
    * Wraps all child processes in one db transaction
    * @param {string} options.path String representing the canonical path for the specified object
-   * @param {string} [options.bucketId] uuid of bucket or `null` if syncing object in default bucket
+   * @param {string | null} options.bucketId uuid of bucket or `null` if syncing object in default bucket
    * @param {boolean} [options.full=false] Optional boolean indicating whether to execute full recursive run
    * @param {string} [options.userId=SYSTEM_USER] Optional uuid attributing which user added the job
    * @returns
    * @throws If the synchronization job encountered an error
    */
-  syncJob: async ({ path, bucketId = undefined, full = false, userId = SYSTEM_USER } = {}) => {
+  syncJob: async ({ path, bucketId, full = false, userId = SYSTEM_USER } = {}) => {
     try {
       if (!path) throw new Error('Path must be defined');
-      const response = [];
-      // start db transaction
-      await utils.trxWrapper(async (trx) => {
 
-        // 1. sync object
-        const object = await service.syncObject({ path: path, bucketId: bucketId, userId: userId }, trx);
+      return await utils.trxWrapper(async (trx) => {
+        const response = [];
+
+        // 1. Sync Object
+        const { newObject, ...object } = await service.syncObject({ path: path, bucketId: bucketId, userId: userId }, trx);
         response.push(object);
 
-        // 2. sync all versions of object
+        // 2. Sync Object Versions
         let versions = [];
-        // IF this is a new (or existing) object in COMS DB (from step 1)
-        // OR doing 'full' sync AND object wasn't deleted
```

Comment:
some kind of comments here explaining the logic are useful. we could add them later.
I think the current general behaviour is that **without** full mode, we only sync versions and meta/tags for **new** objects